src/com/intellij/idea/plugin/hybris/linemaker/HybrisInterceptorLineMakerProvider.java

```diff
@@ -0,0 +1,160 @@
+        }
+        return Collections.emptyList();
+    }
+
+    private boolean isModelClass(final PsiClass psiClass) {
+        final String psiClassName = psiClass.getName();
+        return StringUtils.endsWith(psiClassName, "Model") ||
+               (psiClass.getSuperClass() != null &&
+                psiClass.getSuperClass().getName() != null &&
+                psiClass.getSuperClass().getName().startsWith("Generated"));
+    }
+
+    private boolean isNotHybrisProject(@NotNull final Project project) {
+        return !HybrisProjectSettingsComponent.getInstance(project).getState().isHybrisProject();
+    }
+
+    private boolean filterByGeneric(final PsiClass interceptorPsiClass, final String modelClassName) {
+        final PsiClassType[] interfaces = interceptorPsiClass.getImplementsListTypes();
+        if (interfaces.length > 0) {
+            for (final PsiClassType anInterface : interfaces) {
+                for (PsiType parameter : anInterface.getParameters()) {
+                    if (Objects.equals(modelClassName, ((PsiClassReferenceType) parameter).getName())) {
+                        return true;
+                    }
+                }
+            }
+        }
+
+        final PsiClass superClass = interceptorPsiClass.getSuperClass();
+        if (superClass != null && !StringUtils.equals(superClass.getName(), "Object")) {
+            return filterByGeneric(superClass, modelClassName);
+        }
+        return false;
+    }
+
+    private SmartPsiElementPointer<PsiClass> findInterceptorClass(final Project project) {
+        final PsiClass interceptorClass = JavaPsiFacade.getInstance(project).findClass(
+            "de.hybris.platform.servicelayer.interceptor.Interceptor",
+            GlobalSearchScope.allScope(project)
+        );
+        if (interceptorClass != null) {
+            return SmartPointerManager.getInstance(project).createSmartPsiElementPointer(interceptorClass);
+        }
+        return null;
+    }
+
+    private void createTargetsWithGutterIcon(
+        final Collection<? super RelatedItemLineMarkerInfo<?>> result,
+        final PsiClass psiClass,
+        final Collection<PsiClass> list
+    ) {
+        final NavigationGutterIconBuilder<PsiElement> builder
+            = NavigationGutterIconBuilder.create(HybrisIcons.INTERCEPTOR).setTargets(list);
+
+        builder.setEmptyPopupText(HybrisI18NBundleUtils.message(
+            "hybris.gutter.navigate.no.matching.interceptors"
+        ));
+
+        builder.setPopupTitle(HybrisI18NBundleUtils.message(
+            "hybris.gutter.interceptor.class.navigate.choose.class.title"
+        ));
+        builder.setTooltipText(HybrisI18NBundleUtils.message(
+            "hybris.gutter.interceptor.class.tooltip.navigate.declaration"
+        ));
+        final PsiIdentifier nameIdentifier = psiClass.getNameIdentifier();
+        final RelatedItemLineMarkerInfo<?> lineMarkerInfo = builder.createLineMarkerInfo(nameIdentifier);
```

Comment:
<img width="1270" alt="image" src="https://user-images.githubusercontent.com/2292510/197382237-8b5d54c2-03b8-4f66-9ad2-30f7a0e4b719.png">

mby smth like that:
```
final PsiIdentifier nameIdentifier = psiClass.getNameIdentifier();
        if (nameIdentifier == null) {
            return;
        }

        final NavigationGutterIconBuilder<PsiElement> builder = NavigationGutterIconBuilder.create(HybrisIcons.INTERCEPTOR).setTargets(list);
        final RelatedItemLineMarkerInfo<?> lineMarkerInfo = builder
            .setEmptyPopupText(HybrisI18NBundleUtils.message("hybris.gutter.navigate.no.matching.interceptors"))
            .setPopupTitle(HybrisI18NBundleUtils.message("hybris.gutter.interceptor.class.navigate.choose.class.title"))
            .setTooltipText(HybrisI18NBundleUtils.message("hybris.gutter.interceptor.class.tooltip.navigate.declaration"))
            .createLineMarkerInfo(nameIdentifier);

        result.add(lineMarkerInfo);
```